recipes/pcapplusplus/all/conanfile.py

```diff
@@ -58,15 +77,36 @@ def _configure_sh_script(self):
         }[str(self.settings.os)]
 
     def _patch_sources(self):
-        if not self.options.get_safe("fPIC"):
-            tools.replace_in_file(os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
+        if not self.options.get_safe("fPIC") and self.settings.os != "Windows":
+            files.replace_in_file(self, os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
                                   "-fPIC", "")
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+            files.patch(self, **patch)
 
     def build(self):
         self._patch_sources()
-        with tools.chdir(self._source_subfolder):
+        if self.settings.os == "Windows":
+            self._build_windows()
+        else:
+            self._build_posix()
+
+    def _build_windows(self):
+        vs_version = self._get_vs_version()
+        with files.chdir(self, self._source_subfolder):
+            config_args = [
+                "configure-windows-visual-studio.bat",
+                "--pcap-sdk", self.deps_cpp_info["npcap"].rootpath,
+                "--pthreads-home", self.deps_cpp_info["pthreads4w"].rootpath,
+                "--vs-version", vs_version,
+            ]
+            self.run(" ".join(config_args), run_environment=True)
+            msbuild = MSBuild(self)
+            targets =  ['Common++', 'Packet++', 'Pcap++']
+            cmd = msbuild.command(f"mk/{vs_version}/PcapPlusPlus.sln", targets)
```

Comment:
@uilianries Whatever what the linter is complaining about, I don't see it

Reply:
The problem is that you need to build according to targets, which is not available yet, but it will be available on next release: https://github.com/conan-io/conan/pull/11968

Reply:
@kayoub5 I think you mixing the helpers ... There are 2 MSBuild helper, is almost deprecated, but it seems like fits your current usage, and there is a new and better: https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#msbuild

Here is a template using the new version: https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#msbuild

Tell me if you have questions, I can open a PR to your fork, trying to update your PR with that new MSBuild.

Reply:
@uilianries MSBuild from `conan.tools.microsoft` seems extremely lacking compared to the one from `conans.client.build.msbuild`, for example: does not set correct PlatformToolset or perform project upgrade.

Reply:
@kayoub5 Could you please open an issue to [Conan client](https://github.com/conan-io/conan/issues/), elaborating what is missing? I opened one issue related to MSBuild target, which should include on the next release, but it seems like your case requires one more feature.

Reply:
They already exist:
* https://github.com/conan-io/conan/issues/7419
* https://github.com/conan-io/conan/issues/11967